### PR TITLE
add unittests for the new signal representation

### PR DESCRIPTION
**Pull request by dbuse**
**Currently Work in Progress -- may still change**
Bring the new Signal representation under test coverage. Most tests originally by @f-bronner.
Extracted some common helper functions into separate files.

### CODE REVIEW

File: subprojects/veins_catch/src/SignalRepresentation.cc [hunk 1]
```diff
@@ -0,0 +1,954 @@
...
+        }
+    }
+}
+
+SCENARIO("Signal Timing", "[toolbox]")
+{
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3,4) and a signal (4,3,2,1)")
+    {
+        Freqs freqs = {1, 2, 3, 4};
```
Comment (sommer) [inline subprojects/veins_catch/src/SignalRepresentation.cc]: This is most likely too small for the `signal[4]` that is written to later on.
```suggestion
        Freqs freqs = {1, 2, 3, 4, 5, 6};
```

Comment (dbuse) [reply subprojects/veins_catch/src/SignalRepresentation.cc]: Thanks @sommer for the hint. This actually lead to the root of a bigger problem: Signal allowed silent out-of-bound access to the underlying array. I'm adding some asserts and tests for that.

File: subprojects/veins_catch/src/SignalRepresentation.cc [hunk 2]
```diff
@@ -0,0 +1,983 @@
...
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3) and values for a signal with more frequencies (4,3,2,1)")
+    {
+        Freqs freqs = {1, 2, 3};
+        std::vector<double> values = {4, 3, 2, 1};
+        SpectrumPtr spectrum = Spectrum::getInstance(freqs);
+        Signal signal(spectrum);
+        WHEN("An invalid frequency index of the signal gets read")
+        {
+            THEN("An error should be raised (when in DEBUG mode)")
```
Comment (sommer) [inline subprojects/veins_catch/src/SignalRepresentation.cc]:
```suggestion
#ifndef NDEBUG
            THEN("An error should be raised (when in DEBUG mode)")
```

Comment (dbuse) [reply subprojects/veins_catch/src/SignalRepresentation.cc]: Good idea! Though I'll move the `#ifdef` to embrace the whole scenario.

Catch2 would also support to ignore tests that are expected to throw with the `-e` option (see [here](https://github.com/catchorg/Catch2/blob/master/docs/command-line.md#eliding-assertions-expected-to-throw)). But that would require different calls for debug and release mode runs. Thus I think `#ifndef DEBUG` is the better choice.

**Pull request merged**